controllers/volumereplicationgroup_controller.go

```diff
@@ -617,14 +617,10 @@ func (v *VRGInstance) reconcileVRsForDeletion() bool {
 		pvcNamespacedName := types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}
 		log := v.log.WithValues("pvc", pvcNamespacedName.String())
 
-		requeueResult, skip := v.preparePVCForVRProtection(pvc, log)
-		if requeueResult {
-			requeue = true
-
-			continue
-		}
+		// Skip unprotected PVCs during delete
+		if pvc.Annotations[pvcVRAnnotationProtectedKey] != pvcVRAnnotationProtectedValue {
```

Comment:
Having some doubts on what to check to check to ignore a PVC, I think it should be the finalizer and not the annotation here. (@BenamarMk @raghavendrabhat )

Reply:
So the Annotation guarantees that even the PV metadata has been uploaded to s3.  Whereas the Finalizer doesn't guarantee that. However, it's error-prone if we use the annotation to check for VRProtection.  Because, if we fail to upload to s3 store, then the VRAnnotation will not be set. Hence, we skip reconciling VR for deletion.  so I think it is a good catch @ShyamsundarR.